### PR TITLE
Add test cases for Add opcode implementation

### PR DESCRIPTION
**Pull request by Phantomical**
As in title. Also includes an implementation of `Context::stack_top` and a test case for `evaluate_constant`.

### CODE REVIEW

File: tests/opcodes.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
+
```
Comment (mishazharov) [inline tests/opcodes.cpp:1]: Maybe it makes sense to rename this file to the opcode you're testing? Like `opcode_add` or something like that

Comment (Phantomical) [reply tests/opcodes.cpp:1]: TBH I figured that we'd throw all of those types of tests into a single file so we can centralize the helper methods. It it proves to be too unwieldy then we can split it up but I don't think it's necessary now.

**Review by Skrinikov — APPROVED** [review]
Small comment otherwise lgtm

File: src/decaf.cpp [hunk 2]
```diff
@@ -156,8 +160,6 @@ namespace decaf {
   }
 
   z3::expr evaluate_constant(z3::context& ctx, llvm::Constant* constant) {
-    auto type = constant->getType();
```
[the remark this answers is not in the document]
Comment (Phantomical) [reply src/decaf.cpp:159]: Yeah. I missed the warning about it being unused when I was submitting the last PR.

**Pull request merged**